### ISSUE TITLE
Narrow typed locale identifiers

### ISSUE THREAD

**Issue by warflash** [nuxt-modules/i18n#2857]
### Describe the feature

`const { locale, locales } = useI18n()`

Currently `locale` is typed as `string` and `locales` is `string[]`.
The suggestion is to use tighter types like `locale -> "en" | "de"` in the case of a config like:
```ts
	i18n: {
		locales: [
			{ code: "en", iso: "en-US", file: "en.json" },
			{ code: "de", iso: "de-DE", file: "de.json" },
		],
	},
```

That would make type safe access to typed records possible. e.g.

```ts
const MIN_SEARCH_LEN_BY_LOCALE = {
	en: 3,
	de: 3,
	ko: 1,
}

const { locale, locales } = useI18n()

const minLen = MIN_SEARCH_LEN_BY_LOCALE[locale] 
```

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution) (The contribution guideline of nuxt-modules/i18n is compliant with Nuxt too).
- [X] Check existing [discussions](https://github.com/nuxt-modules/i18n/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by BobbieGoede** [comment]
Should be resolved by #3025, which will be included in v9

**Comment by warflash** [comment]
Thanks! 💚
